firestore/app/src/main/java/com/google/firebase/example/fireeats/kotlin/FilterDialogFragment.kt

```diff
@@ -122,8 +122,8 @@ class FilterDialogFragment : DialogFragment() {
     override fun onAttach(context: Context) {
         super.onAttach(context)
 
-        if (context is FilterListener) {
-            filterListener = context
+        if (parentFragment is FilterListener) {
+            filterListener = parentFragment as FilterListener
```

Comment:
nit: I wonder if Kotlin's smart casting would work here?
It's just a personal curiosity, it might not work.
```suggestion
            filterListener = parentFragment
```

Reply:
It did not work here, I tried that. I think if I captured parentFragment with a `let` then it might.